primenet.py

```diff
@@ -4795,14 +4795,18 @@ def get_assignments(adapter, adir, cpu_num, progress, tasks):
         time_left = timedelta(seconds=time_left)
         days_work = timedelta(days=options.days_of_work)
         if time_left <= days_work:
-            num_cache += 1
+            num_cache = math.ceil(num_existing * days_work / time_left)
```

Comment:
This is perhaps the "exact" calculation of how big our cache should be, assuming the time_left estimation is correct, and all jobs are similarly sized. (and should work for every worktype)

The drawbacks of doing it this way are that if we start up a fresh instance, we won't have a time calculation, so we have to choose some amount of assignments for the first run, in the case that the user hasn't used num_cache. I've chosen 10 for TF and 1 for other types (1 was the previous behavior in this case, I believe)

The other drawback is that we might accidentally fetch the user a large amount of assignments, if they have an underestimate for time_left or something. Some guard-rails would maybe be prudent, but I can't think of an easy way to do it that won't be annoying when someone is trying to fetch 10 days of work before taking their computer immediately offline.

After all, the user could then --unreserve-all if we accidentally fetch too much.

Reply:
10 assignments would hold over my GPU for roughly two hours, so anything above that for --timeout would result in mfaktc closing before more work received. Perhaps in the "fresh" case, we want to get a small amount of assignments, and wake every minute until we have a good time estimate, and then get more work?

Not sure what the most general solution to this is, I don't necessarily want to have a bunch of carveouts for TF if we don't have to.

Reply:
If the `--num-cache` option is larger than what is needed for the `--days-work` option, this will select the smaller of the two values. I think you want something like this instead (which also eliminates the `math.ceil`):
```suggestion
            num_cache = max(num_cache, -(days_work * num_existing // -time_left))
```

Reply:
Good catch, I've redone the debug lines relating to this to make a bit more sense in the case where num_cache is bigger than days_of_work calculates is necessary.